sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/View.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.connector.catalog;
+
+import java.util.Map;
+
+import org.apache.spark.annotation.DeveloperApi;
+import org.apache.spark.sql.types.StructType;
+
+/**
+ * An interface representing a persisted view.
+ */
+@DeveloperApi
+public interface View {
+  /**
+   * A name to identify this view.
+   */
+  String name();
+
+  /**
+   * The view query SQL text.
+   */
+  String sql();
+
+  /**
+   * The current catalog when the view is created.
+   */
+  String currentCatalog();
+
+  /**
+   * The current namespace when the view is created.
+   */
+  String[] currentNamespace();
+
+  /**
+   * The schema for the SQL text when the view is created.
+   */
+  StructType schema();
+
+  /**
+   * The view column aliases.
+   */
+  String[] columnAliases();
+
```

Comment:
Could you please clarify how this method is being used? By checking the umbrella PR, (#28147), it does not look like it is used either.

Reply:
This is to capture this part in Spark's CREATE VIEW syntax:
```
( { column_alias [ COMMENT column_comment ] } [, ...] )
```
Array columnAliases and columnComments must have the same size.

#28147 does not have it but we need to add them to maintain existing Spark features.

Reply:
@jzhuge is the intent that `schema()` contains the _pre-alias_ schema and `columnAliases()` contains the list of aliases, such that you need to combine `schema() + columnAliases()` to get the _post-alias_ schema? This is in contrast to how V1 views work, where `CatalogTable.schema()` contains the _post-alias_ schema and `viewQueryColumnNames` contains the original column names. The plan is constructed by aliasing from the `viewQueryColumnNames` to the names stored in `schema` (see [this code in SessionCatalog](https://github.com/apache/spark/blob/933dc0c42f0caf74aaa077fd4f2c2e7208452b9b/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala#L949-L985)).

I don't have a strong opinion about whether it makes more sense for `schema()` to be pre- or post-aliasing, but we need to make this very clear in the interface description. If we go with pre-aliasing, we can convert it to post-alias inside of [V2ViewDescription](https://github.com/apache/spark/pull/28147/files#diff-d0c6f499b30df039d13bf2740c559251ab63ba4cea312e622e23b74fcbb2fcf0) to make it match V1 in the analyzer, reducing implementation complexity by avoiding having to special-case V1 vs V2.

Reply:
No, `schema()` contains the final schema, the same as V1, to reduce confusion. I will update javadoc to clarify.

Thank you and @wmoustafa for calling it out!

For V1, schema stores column aliases. But for V2, we decided to use these metadata fields to capture the entire CREATE VIEW statement instead of relying on `schema` which feels more like "derived":
- sql
- columnAliases
- columnComments

Let me know if this makes sense.

Reply:
If both `columnAliases` and `schema` contain the final output names (post-alias), then where are the original column names stored? We need those to reconstruct the aliasing similar to the code I linked from `SessionCatalog`. Each element of the projection looks like `Alias(UpCast(<originalColName>, <outputDataType>), <alias>)`, which cannot be constructed without access to the original column names.

Reply:
Yeah, for select star, we need something similar to `viewQueryColumnNames`.

If schema is pre-alias, then `schema.fieldNames` can serve the purpose.

Reply:
Pushed a few changes:
- Updated Javadoc for View.schema to indicate aliases applied
- Added `View.queryColumnNames` to store output column names, useful for SELECT star queries
- Renamed `View.sql` to `View.query` because "query" is more accurate to describe the SELECT query for the view.

Reply:
LGTM. We are capturing some redundant information but I recognize the value in trying to persist everything from the `CREATE VIEW` statement to make things clear and not rely on derived attributes.